src/Send/Memo.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {HelperText, TextInput} from '../components'
+
+export const maxMemoLength = 256
+
+type Props = {
+  memo: string
+  onChangeText: (memo: string) => void
+}
+
+export const Memo = ({onChangeText, memo}: Props) => {
+  const strings = useStrings()
+
+  const showError = memo.length > maxMemoLength
+  const multilineMax = memo.length > 50
+
+  return (
+    <View style={styles.container}>
+      <TextInput
+        value={memo}
+        onChangeText={(memo) => onChangeText(memo.trim())}
+        label={strings.label}
+        autoComplete={false}
+        testID="memoFieldInput"
+        errorText={showError ? 'error' : undefined} // to show the error styling
+        noHelper={true}
+        style={multilineMax && styles.input}
+        multiline={true}
+      />
+
+      <View style={styles.helper}>
```

Comment:
I've finally opted for an intermediate solution for a reason:

The TextInput helper has several dependencies on TextInput and has a specific style

Reply:
Let me know if you think otherwise